### PR TITLE
fix(docusaurus 3): update docs to be mdx v3 compliant

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

Part of #2719.

Updates many small syntax issues in the docs, to be valid MDX (v3). 

Guidance provided by https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3 and https://mdxjs.com/docs/troubleshooting-mdx/, and tip of the hat to https://mdxjs.com/playground/ for providing a much faster/tighter feedback loop than the docusaurus checker when debugging issues. 

The issues fall into these basic categories:

- Elements that open on one line, include text before the end of line, then close on a different line. (Fixed by moving the text to a new line.)
- code that wasn't living in a fenced codeblock (shows as acorn parsing errors)
- unclosed HTML tags
- unescaped brackets/braces

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [x] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. 
   **My goal is to upgrade to docusaurus 3 this week, so I'd like to merge these changes in the next day.**
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

- [x] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process).

### CODE REVIEW

File: docs/components/best-practices/development/routing-events-to-processes-c7.md [hunk 1]
```diff
@@ -327,10 +327,11 @@ public class InvoiceMDB implements MessageListener {
 
 The provided REST API can be directly used to communicate with the workflow engine remotely.
 
-```
 POST /process-definition/key/invoice/start
```
Comment (akeller) [inline docs/components/best-practices/development/routing-events-to-processes-c7.md:330]: I think we _do_ want some of this info in a code block, but not the same one.

File: docs/components/modeler/desktop-modeler/telemetry/telemetry.md [hunk 2]
```diff
@@ -54,8 +54,8 @@ These events include the following properties:
 
 - `diagramType`: BPMN, DMN, or Form
 - Engine profile:
-  - `executionPlatform`: <target platform\>
-  - `executionPlatformVersion`: <target platform version\>
+  - `executionPlatform`: &lt;target platform\>
```
Comment (akeller) [inline docs/components/modeler/desktop-modeler/telemetry/telemetry.md:57]: Interesting... something to take to the tech writers about how we want to format info like this.

**Review by akeller — APPROVED** [review]
Approving. I left a few comments where I did some spot checking, including one about code block. I couldn't quickly find a good example of what we do elsewhere in the docs, so I considered this non-blocking.

**Comment by akeller** [comment]
> The issues fall into these basic categories:
> 
> * Elements that open on one line, include text before the end of line, then close on a different line. (Fixed by moving the text to a new line.)
> * code that wasn't living in a fenced codeblock (shows as acorn parsing errors)
> * unclosed HTML tags
> * unescaped brackets/braces

Thank you for this! 🧡

File: docs/components/modeler/desktop-modeler/telemetry/telemetry.md (hunk 2 above)
Comment (pepopowitz) [reply docs/components/modeler/desktop-modeler/telemetry/telemetry.md:57]: Yeah...we are not universal in how we present "a value that is represented by something we can describe". Quotes, curly braces, angle brackets....those seem to be the three ways I can remember seeing.

**Pull request merged**